### PR TITLE
2.2 fixes

### PR DESCRIPTION
**Pull request by Ritesh-patel**
- Fix tax_query "OR" relation support in EP API
- Use ep_elasticsearch_can_connect() instead of ep_elasticsearch_alive()

### CODE REVIEW

**Comment by Ritesh-patel** [comment]
Fixes #632

File: elasticpress.php [hunk 1]
```diff
@@ -55,37 +55,46 @@
...
 
 /**
- * On activate, all features that meet their requirements with no warnings should be activated.
+ * All features that meet their requirements with no warnings should be activated.
  *
- * @since  2.1
+ * @since 2.2
  */
-function ep_on_activate() {
+function ep_auto_activate_features() {
```
Comment (tlovett1) [inline elasticpress.php]: Does it make sense to move this to class-ep-features.php?

Comment (Ritesh-patel) [inline elasticpress.php]: Yes, we can move that as it's feature specific.

**Comment by tlovett1** [comment]
For the auto activate piece, are you just moving the code to another file? If so, let's remove that part which should solve the conflict.

**Comment by Ritesh-patel** [comment]
2 different functions uses the same code so I have moved that code into a wrapper function. PR has been updated accordingly.

**Comment by tlovett1** [comment]
Ritesh please just separate this into multiple PRs. The auto-activate function you created does two completely different things (2nd is setting an upgrade needed option)

**Comment by Ritesh-patel** [comment]
@tlovett1 running the plugin activation code on `plugins_loaded` hook doesn't sound good. I think maintaining an option which can tell whether to upgrade or not is still a valid option. This is the case basically for plugin updates where `register_activation_hook` won't be called.

**Comment by tlovett1** [comment]
Let's table this one until our next meeting so we can discuss.

**Comment by tlovett1** [comment]
I pushed the taxonomy fix as a separate commit as it was breaking tests.

**Pull request closed without merging**